DEV_GUIDE.md

```diff
@@ -26,6 +27,22 @@ This document gives a detailed breakdown of the various build processes and opti
   * [Rendering documentation](#rendering-documentation)
 <!-- TOC -->
 
+## Project structure
+
+The module structure is as follows:
+
+* `kroxylicious-bom` the Bill Of Materials used by developers to conveniently declare a consistent set of Kroxylicious dependencies.
+* `krpc-code-gen` a code generator used to build the API module.
+* `kroxylicious-api` the APIs needed to write a filter.
+* `kroxylicious-unit-test` test utilities for filter authors.
+* `kroxylicious` the proxy runtime platform for hosting filters, including the networking code etc, but lacking a `main()` method
+* `kroxylicious-test-tools` ???
+* `kroxylicious-app` provides a `main()` method for the proxy.
+* `kroxylicious-multitenant` a filter that implements multitenancy
+* `kroxylicious-schema-enforcement` a filter that implements schema enforcement
+* `integrationtests` the integration test suite
+* `kroxylicious-sample` contains a sample filter
```

Comment:
The order of the bullets is the same as the order maven executes the poms, i.e. dependency order. It's not clear to me why integrationtests should come first. In principal we should be able to integration test our sample.

Reply:
We do [integration test](https://github.com/kroxylicious/kroxylicious/blob/04a1e8dc1cae8da73ef7903ae1669ce272d3c8d8/kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFilterIntegrationTest.java) the sample, we just don't do it as part of the project integration test suite as that wouldn't be an option for external filter authors.